analytics/adobe-analytics/experience-accelerator/src/AAClient.ts

```diff
@@ -1,22 +1,32 @@
 import { Client } from "./Client";
+type Dimensions = "session" | "user" | "candidate" | "group" | "ordinal"
+type DimensionsValue = { key: string, type?: string };
+type DimensionsMap = { [dimension in Dimensions]?: DimensionsValue};
 
 export class AAClient extends Client {
     constructor(
-        public readonly sessionIdDimension: string,
-        public readonly userIdDimension: string,
-        public readonly candidateIdDimension: string,
-        public readonly sessionIdDimensionType = 'eVar',
-        public readonly userIdDimensionType = 'eVar',
-        public readonly candidateIdDimensionType = 'eVar',
+        public readonly dimensions: DimensionsMap,
```

Comment:
How many clients are currently using AA? This will break their implementation and need some coordination to update their implementation.